packages/datagateway-table/src/breadcrumbs.component.tsx

```diff
@@ -0,0 +1,538 @@
+    url: string;
+    isLast: boolean;
+  };
+  currentHierarchy: Breadcrumb[];
+  last: {
+    displayName: string;
+  };
+}
+
+interface WrappedProps extends WithStyles<typeof styles> {
+  displayName: string;
+  ariaLabel: string;
+  url?: string;
+  isLast?: boolean;
+}
+
+class WrappedBreadcrumb extends React.Component<
+  WrappedProps & WithStyles<typeof styles>
+> {
+  public constructor(props: WrappedProps & WithStyles<typeof styles>) {
+    super(props);
+  }
+
+  public render(): React.ReactElement {
+    return (
+      <ArrowTooltip title={this.props.displayName}>
+        {this.props.url ? (
+          <MaterialLink
+            component={Link}
+            to={this.props.url}
+            aria-label={this.props.ariaLabel}
+            className={this.props.classes.breadcrumb}
+          >
+            {this.props.displayName}
+          </MaterialLink>
+        ) : (
+          <Typography
+            color="textPrimary"
+            aria-label={this.props.ariaLabel}
+            className={this.props.classes.breadcrumb}
+          >
+            {this.props.isLast ? (
+              <i>{this.props.displayName}</i>
+            ) : (
+              this.props.displayName
+            )}
+          </Typography>
+        )}
+      </ArrowTooltip>
+    );
+  }
+}
+
+const StyledBreadcrumb = withStyles(styles)(WrappedBreadcrumb);
+
+class PageBreadcrumbs extends React.Component<
+  BreadcrumbProps,
+  BreadcrumbState
+> {
+  private currentPathnames: string[];
+  private isBreadcrumbUpdated: boolean;
+
+  public constructor(props: BreadcrumbProps) {
+    super(props);
+
+    this.currentPathnames = [];
+    this.isBreadcrumbUpdated = false;
+
+    this.state = {
+      base: {
+        entityName: '',
+        displayName: 'N/A',
+        url: '',
+        isLast: false,
+      },
+      currentHierarchy: [],
+      last: {
+        displayName: 'N/A',
+      },
+    };
+  }
+
+  public componentDidMount(): void {
+    // Store the current pathnames.
+    this.currentPathnames = this.props.location.split('/').filter(x => x);
+    this.updateBreadcrumbState();
+  }
+
+  public componentDidUpdate(prevProps: BreadcrumbProps): void {
+    this.currentPathnames = this.props.location.split('/').filter(x => x);
+
+    // TODO: Do we need isBreadcrumbUpdated anymore?
+    if (prevProps.location !== this.props.location)
+      this.isBreadcrumbUpdated = false;
+
+    if (!this.isBreadcrumbUpdated) {
+      this.updateBreadcrumbState();
+    }
+  }
+
+  private updateBreadcrumbState = async () => {
+    let updatedState = this.state;
+
+    // Update the base address if it has changed.
+    const baseEntityName = this.currentPathnames[1];
+    if (!(updatedState.base.entityName === baseEntityName)) {
+      updatedState = {
+        ...updatedState,
+        base: {
+          entityName: baseEntityName,
+          displayName:
+            `${baseEntityName}`.charAt(0).toUpperCase() +
+            `${baseEntityName}s`.slice(1),
```

Comment:
There is an internationalisation issue here - [reference internationalisation issue]